src/rgw/rgw_op.cc

```diff
@@ -5949,6 +5949,39 @@ void RGWCompleteMultipart::execute(optional_yield y)
   op_ret = serializer->try_lock(dur, y);
   if (op_ret < 0) {
     ldpp_dout(this, 0) << "failed to acquire lock" << dendl;
+    if (op_ret == -ENOENT) {
+      std::string oid = mp.get_part(1);
+      rgw_obj chk_obj;
+      chk_obj.init_ns(s->bucket->get_key(), oid, mp_ns);
+      rgw_obj_index_key key;
+      chk_obj.key.get_index_key(&key);
+      bool bret;
+      rgw_pool pool;
+      bret = static_cast<rgw::sal::RGWRadosStore*>(store)->getRados()->get_obj_data_pool(s->bucket->get_placement_rule(), chk_obj, &pool);
```

Comment:
isn't this problematic?  this causes complete-multipart to require RADOS
@dang how should Mark proceed here?

Reply:
This can go in as-is to facilitate backports, and I'll re-work it by adding a mpstat() API (or something like it) as part of the module refactoring work.

Reply:
i'd prefer not to crash here once we do add a different backend. can we at least use `dynamic_cast` so we can check for null? and add a TODO comment about exposing something like mpstat() in zipper

```c++
auto rados = dynamic_cast<rgw::sal::RGWRadosStore*>(store);
if (!rados) { -> ERR_INTERNAL_ERROR
```